### PR TITLE
very minor changes to make it run on Windows.

### PR DESCRIPTION
**Pull request by SamGG**

### CODE REVIEW

File: src/Makevars [hunk 1]
```diff
@@ -2,7 +2,7 @@ CXX_STD = CXX11
 
 PKG_CPPFLAGS = -I../inst/include/
 PKG_CXXFLAGS = $(SHLIB_OPENMP_CXXFLAGS) -DNO_MANUAL_VECTORIZATION -DSTRICT_R_HEADERS
-PKG_LIBS = $(SHLIB_OPENMP_CXXFLAGS) $(BLAS_LIBS)
```
Comment (dselivanov) [inline src/Makevars:5]: Why removing BLAS is needed? despite we don't use BLAS routines in this pkg, I think it still should compile fine.

File: src/hnsw.cpp [hunk 2]
```diff
@@ -116,8 +116,8 @@ class Hnsw {
     // we can't touch R API in a threaded code, but we can use data
     // allocated by R
     // so we need to get a pointer to the underlying matrix data
-    uint32_t ncol = items.ncol();
-    uint32_t nrow = items.nrow();
+    int ncol = items.ncol();
```
Comment (dselivanov) [inline src/hnsw.cpp:119]: could you rather try to add `#include <stdint.h>` and see if it helps?

**Review by dselivanov — CHANGES_REQUESTED** [review]

File: src/Makevars (hunk 1 above)
Comment (SamGG) [reply src/Makevars:5]: This raises a warning
```
> checking for portable use of $(BLAS_LIBS) and $(LAPACK_LIBS) ... WARNING
    apparently using $(BLAS_LIBS) without following $(FLIBS) in 'src/Makevars'
```

File: src/hnsw.cpp (hunk 2 above)
Comment (SamGG) [reply src/hnsw.cpp:119]: It does not seem to help removing the warnings
```
   "C:/rtools40/mingw64/bin/"g++  -std=gnu++11 -I"C:/opt/R/R-40~1.0/include" -DNDEBUG -I../inst/include/ -I'C:/opt/R/R-4.0.0/library/Rcpp/include'     -fopenmp -DNO_MANUAL_VECTORIZATION -DSTRICT_R_HEADERS   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -UNDEBUG -Wall -pedantic -g -O0 -fdiagnostics-color=always -c hnsw.cpp -o hnsw.o
   hnsw.cpp: In instantiation of 'void Hnsw<dist_t, Distance, DoNormalize>::addItems(Rcpp::NumericMatrix, int) [with dist_t = float; Distance = hnswlib::L2Space; bool DoNormalize = false; Rcpp::NumericMatrix = Rcpp::Matrix<14>]':
   hnsw.cpp:329:34:   required from here
   hnsw.cpp:132:15:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
          #pragma omp for schedule(dynamic, 64)
                  ^~~
   hnsw.cpp:136:28:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
            for (auto j = 0; j < ncol; j ++) {
                             ~~^~~~~~
   hnsw.cpp: In instantiation of 'void Hnsw<dist_t, Distance, DoNormalize>::addItems(Rcpp::NumericMatrix, int) [with dist_t = float; Distance = hnswlib::InnerProductSpace; bool DoNormalize = true; Rcpp::NumericMatrix = Rcpp::Matrix<14>]':
   hnsw.cpp:349:38:   required from here
   hnsw.cpp:132:15:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
          #pragma omp for schedule(dynamic, 64)
                  ^~~
   hnsw.cpp:136:28:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
            for (auto j = 0; j < ncol; j ++) {
                             ~~^~~~~~
   hnsw.cpp: In instantiation of 'void Hnsw<dist_t, Distance, DoNormalize>::addItems(Rcpp::NumericMatrix, int) [with dist_t = float; Distance = hnswlib::InnerProductSpace; bool DoNormalize = false; Rcpp::NumericMatrix = Rcpp::Matrix<14>]':
   hnsw.cpp:369:36:   required from here
   hnsw.cpp:132:15:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
          #pragma omp for schedule(dynamic, 64)
                  ^~~
   hnsw.cpp:136:28:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
            for (auto j = 0; j < ncol; j ++) {
                             ~~^~~~~~
   C:/rtools40/mingw64/bin/g++ -shared -s -static-libgcc -o RcppHNSW.dll tmp.def RcppExports.o hnsw.o -fopenmp -LC:/opt/R/R-40~1.0/bin/x64 -lRblas -LC:/opt/R/R-40~1.0/bin/x64 -lR
   installing to C:/Users/samgg/AppData/Local/Temp/Rtmp2X6kjo/devtools_install_137c171577c3/00LOCK-rcpphnsw/00new/RcppHNSW/libs/x64
-  DONE (RcppHNSW)
```

File: src/Makevars (hunk 1 above)
Comment (dselivanov) [reply src/Makevars:5]: alright, removing `$(BLAS_LIBS)`

File: src/hnsw.cpp (hunk 2 above)
Comment (dselivanov) [reply src/hnsw.cpp:119]: Sorry, just want to clarify. Do you mean that you have `uint32_t nrow, ncol` and still you have warning "comparison of integer expressions of different signedness"  in the for loops?

Comment (SamGG) [reply src/hnsw.cpp:119]: No problem to clarify. When I set uint32_t for ncol and nrow, I got the warnings above. When I set int, there are no warnings at all.

Comment (dselivanov) [reply src/hnsw.cpp:119]: strange! =)

**Comment by dselivanov** [comment]
Thanks for PR

**Comment by SamGG** [comment]
Thanks to you for providing the core of this interesting improvement.
Hope to work with you again.
Best.

**Comment by SamGG** [comment]
For my training, why `schedule(dynamic, 64)` instead of `schedule(static)`?

**Comment by dselivanov** [comment]
not much difference in this case. dynamic is useful when each job can vary significantly in terms of time required to finish

**Pull request merged**